src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/CoverageLogic.kt

```diff
@@ -42,6 +43,34 @@ class RepositoriesCoverageLogic(private val modellingGroupRepository: ModellingG
                 scenarioAndCoverageSets.coverageSets)
     }
 
+    override fun saveCoverageForTouchstone(touchstoneVersionId: String, rows: Sequence<CoverageIngestionRow>)
+    {
+        val setDeterminants = mutableListOf<Triple<ActivityType, GAVISupportLevel, String>>()
+        val setIds = mutableListOf<Int>()
+        val records = rows.map {
+            val set = Triple(it.activityType, it.gaviSupport, it.vaccine)
+            var setIndex = setDeterminants.indexOf(set)
+            if (setIndex == -1)
+            {
+                val newId = touchstoneRepository.createCoverageSet(touchstoneVersionId, it.vaccine, it.activityType, it.gaviSupport)
+                setIds.add(newId)
+                setDeterminants.add(set)
+                setIndex = setIds.count() - 1
+            }
+            val id = setIds[setIndex]
+            touchstoneRepository.newCoverageRowRecord(
```

Comment:
Unlike for burden estimates, where we do something very complicated with streaming and using Postgres COPY, here I'm just doing the most naive thing of building up a list of records and inserting them. The magnitude of this data is significantly less than burden estimates, so I figured lets see how far this approach takes us.